src/api/Controllers/BoreholeController.cs

```diff
@@ -28,6 +28,7 @@ public async override Task<ActionResult<Borehole>> EditAsync(Borehole entity)
         }
 
         var existingBorehole = await Context.Boreholes
+            .Include(b => b.BoreholeCodelists)
```

Comment:
@danjov Fredo und ich hatten gestern ein Problem, dass wenn das Include hier fehlte, der Unittest unten durchlief, aber ein mehrfaches updaten über den client zu Fehlern führte. Wir haben es nicht geschafft, dass Verhalten im Test abzubilden. Hast du grad eine Idee? Transactions? Neuer Context?

Reply:
Hängt wohl irgendwie damit zusammen, dass im Test alles zu einer Transaktion zusammengewurstet wird und real-life 2 request auf die DB gehen. Jetzt geht ja, ist nur blöd, wenn der Test uns nicht die Wahrheit sagt.

2024-11-05 15:02:22           MessageText: duplicate key value violates unique constraint "PK_borehole_identifiers_codelist"
2024-11-05 15:02:22           Detail: Detail redacted as it may contain sensitive data. Specify 'Include Error Detail' in the connection string to include this information.
2024-11-05 15:02:22           SchemaName: bdms
2024-11-05 15:02:22           TableName: borehole_identifiers_codelist
2024-11-05 15:02:22           ConstraintName: PK_borehole_identifiers_codelist

Reply:
@MiraGeowerkstatt Meine Vermutung: Weil der `Include(..)` gefehlt hat, hat EF beim zweiten Durchgang angenommen, die Codeliste sei neu und hat deshalb versucht diese neu zu erstellen bei einem Update. Wenn der `Include(..)` drin ist, wird die Codeliste mit dem Borehole geladen und EF erkennt, ob Codelisten neu oder geändert sind.  
Um dieses Verhalten in einem Test abzubilden, müsste man pro Update einen neuen Context verwenden. Weil der Context im Test wiederverwendet wird, erkennt er die Änderung an der Codeliste und wirft keinen Fehler.

Reply:
@danjov Okay ja, das würde das Verhalten erklären. Wäre cool wenn man den Fehler reproduzieren könnte im Test.

Reply:
@danjov @Lehats habe den Test jetzt angepasst sodass er verschiedene Contexts verwendet pro Aktion :) 
Damit konnte ich den Fehler reproduzieren, allerdings bricht jetzt das Konzept vom kompletten automatischen Rollback der Transaction am Ende des Test mit dem disposen des TestContext. 
Wenn der Test durchläuft ist die DB auch wieder im Zustand wie vorher und das Borehole und die Identifiers werden in keinem anderen Test verwendet. Findet ihr das so vertretbar?

Reply:
@MiraGeowerkstatt Um ganz sicher zu gehen kann man die Testdaten auch _manuell_ nach den Tests in der Klasse abräumen. So gehen wir sicher, dass es in Zukunft nie Seiteneffekte geben wird. Dafür kann man eine Methode mit dem `ClassCleanup`-Attribut versehen, diese Methode wird nach allen Tests in der entsprechenden Klasse ausgeführt.

Reply:
@MiraGeowerkstatt @danjov Für mich passt das mit dieser Lösung.